components/contribution-flow/StepDetails.js

```diff
@@ -129,7 +130,7 @@ const StepDetails = ({ onChange, data, collective, tier, showFeesOnTop, router }
             id="contribute.tierDetails"
             defaultMessage="You’ll contribute {amount}{interval, select, month { monthly} year { yearly} other {}}."
             values={{
-              interval: tier.interval,
+              interval: tier.interval !== INTERVALS.flexible ? tier.interval : null,
```

Comment:
The goal of this line is to describe your contribution. If we display `You’ll contribute €41.98 EUR` (without the interval) while the contribution is a monthly one, it's misleading information.

We should rollback this line, but make sure that we cannot end up in a situation where `isFixedContribution` is true while the interval is flexible. In the tiers form, `Interval` should come before `Amount type` and the latter should be hidden when the interval is set to FLEXIBLE (with its value forced to be flexible as well).